File: node/src/resources/apiKeys.ts
import { TransmitClient } from '../client';
import { unwrapList, unwrapPaginated } from '../response';
import { CreateApiKeyRequest } from '../types';

export class ApiKeys {
  private client: TransmitClient;

  constructor(client: TransmitClient) {
    this.client = client;
  }

  async list(): Promise<any[]> {
    const raw = await this.client.get<unknown>('/api/v1/developers/api-keys');
    return unwrapPaginated(raw).items;
  }

  async create(data: CreateApiKeyRequest): Promise<any> {
    return this.client.post<any>('/api/v1/developers/api-keys', data);
  }

  async retrieve(id: string): Promise<any> {
    return this.client.get<any>(`/api/v1/developers/api-keys/${id}`);
  }

  async revoke(id: string): Promise<void> {
    return this.client.post<void>(`/api/v1/developers/api-keys/${id}/revoke`);
  }

  async delete(id: string): Promise<void> {
    return this.client.delete<void>(`/api/v1/developers/api-keys/${id}`);
  }

  async getUsage(id: string): Promise<any> {
    return this.client.get<any>(`/api/v1/developers/api-keys/${id}/usage`);
  }
}

File: rust/src/resources/api_keys.rs
use crate::client::{Client, TransmitError};
use crate::types::ApiKeyCreateRequest;
use serde_json::Value;

pub struct ApiKeys<'a> {
    client: &'a Client,
}

impl<'a> ApiKeys<'a> {
    pub fn new(client: &'a Client) -> Self {
        Self { client }
    }

    pub async fn list(&self) -> Result<Vec<Value>, TransmitError> {
        self.client.get("/api/v1/developers/api-keys").await
    }

    pub async fn create(&self, req: &ApiKeyCreateRequest) -> Result<Value, TransmitError> {
        self.client.post("/api/v1/developers/api-keys", req).await
    }

    pub async fn retrieve(&self, id: &str) -> Result<Value, TransmitError> {
        let path = format!("/api/v1/developers/api-keys/{}", id);
        self.client.get(&path).await
    }

    pub async fn revoke(&self, id: &str) -> Result<Value, TransmitError> {
        let path = format!("/api/v1/developers/api-keys/{}/revoke", id);
        self.client.post(&path, &serde_json::json!({})).await
    }

    pub async fn delete(&self, id: &str) -> Result<Value, TransmitError> {
        let path = format!("/api/v1/developers/api-keys/{}", id);
        self.client.delete(&path).await
    }

    pub async fn get_usage(&self, id: &str) -> Result<Value, TransmitError> {
        let path = format!("/api/v1/developers/api-keys/{}/usage", id);
        self.client.get(&path).await
    }
}
